Purchases/Networking/Operations/GetIntroEligibilityOperation.swift

```diff
@@ -0,0 +1,133 @@
+         completion: @escaping IntroEligibilityResponseHandler) {
+        self.configuration = configuration
+        self.receiptData = receiptData
+        self.productIdentifiers = productIdentifiers
+        self.completion = completion
+
+        super.init(configuration: configuration)
+    }
+
+    override func main() {
+        if self.isCancelled {
+            return
+        }
+
+        self.getIntroEligibility(appUserID: self.configuration.appUserID,
+                                 receiptData: self.receiptData,
+                                 productIdentifiers: self.productIdentifiers,
+                                 completion: self.completion)
+    }
+
+    func getIntroEligibility(appUserID: String,
+                             receiptData: Data,
+                             productIdentifiers: [String],
+                             completion: @escaping IntroEligibilityResponseHandler) {
+        guard productIdentifiers.count > 0 else {
+            completion([:], nil)
+            return
+        }
+
+        if receiptData.count == 0 {
+            if self.httpClient.systemInfo.isSandbox {
+                Logger.appleWarning(Strings.receipt.no_sandbox_receipt_intro_eligibility)
+            }
+
+            var eligibilities: [String: IntroEligibility] = [:]
+
+            for productID in productIdentifiers {
+                eligibilities[productID] = IntroEligibility(eligibilityStatus: .unknown)
+            }
+
+            completion(eligibilities, nil)
+            return
+        }
+
+        // Closure we can use for both missing appUserID as well as server error where we have an unknown
+        // eligibility status.
+        let unknownEligibilityClosure: () -> [String: IntroEligibility] = {
+            let unknownEligibilities = [IntroEligibility](repeating: IntroEligibility(eligibilityStatus: .unknown),
+                                                          count: productIdentifiers.count)
+            let productIdentifiersToEligibility = zip(productIdentifiers, unknownEligibilities)
+            return Dictionary(uniqueKeysWithValues: productIdentifiersToEligibility)
+        }
+
+        guard let appUserID = try? appUserID.escapedOrError() else {
+            completion(unknownEligibilityClosure(), ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let fetchToken = receiptData.asFetchToken
+        let path = "/subscribers/\(appUserID)/intro_eligibility"
+        let body: [String: Any] = ["product_identifiers": productIdentifiers,
+                                   "fetch_token": fetchToken]
+
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, maybeResponse, error in
+            let eligibilityResponse = IntroEligibilityResponse(maybeResponse: maybeResponse,
```

Comment:
this feels similar to the `createAliasResponseHandler` above. wondering if there's a way to unify the respond handling methodology... 🤔

Reply:
I'm going to call it out of scope. Can do a follow up to explore this.